interfaces/builtin/overmount.go

```diff
@@ -0,0 +1,156 @@
+	return nil, nil
+}
+
+func (iface *OvermountInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+// path is an internal helper that extract the "read" and "write" attribute
+// of the slot
+func (iface *OvermountInterface) path(slot *interfaces.Slot, name string) []string {
+	if name != "read" && name != "write" && name != "execute" {
+		panic("internal error, path can only be used with read/write/execute")
+	}
+
+	paths, ok := slot.Attrs[name].([]interface{})
+	if !ok {
+		return nil
+	}
+
+	out := make([]string, len(paths))
+	for i, p := range paths {
+		out[i], ok = p.(string)
+		if !ok {
+			return nil
+		}
+	}
+	return out
+}
+
+func (iface *OvermountInterface) overmountMountSnippet(plug *interfaces.Plug, slot *interfaces.Slot) ([]byte, error) {
+	snippet := bytes.NewBuffer(nil)
+	for _, r := range iface.path(slot, "read") {
+		fmt.Fprintln(snippet, mountEntry(plug, slot, r, ",ro"))
+	}
+	for _, w := range iface.path(slot, "write") {
+		fmt.Fprintln(snippet, mountEntry(plug, slot, w, ""))
+	}
+	for _, e := range iface.path(slot, "execute") {
+		fmt.Fprintln(snippet, mountEntry(plug, slot, e, ",exec"))
+	}
+	return snippet.Bytes(), nil
+}
+
+func appArmorEntry(plug *interfaces.Plug, slot *interfaces.Slot, relSrc, mntOpts, permissions string) string {
+	/* dst */ _ = resolveSpecialVariable(plug.Attrs["target"].(string), plug.Snap)
+	/* src */ _ = resolveSpecialVariable(relSrc, slot.Snap)
+	return fmt.Sprintf("mount options=(ro bind %s) %s/*/** -> %s/** %s,", "/snap", "exec", "/usr/bin", permissions)
+}
+
+func (iface *OvermountInterface) overmountAppArmorSnippet(plug *interfaces.Plug, slot *interfaces.Slot) ([]byte, error) {
+	snippet := bytes.NewBuffer(nil)
+	for _, r := range iface.path(slot, "read") {
+		fmt.Fprintln(snippet, appArmorEntry(plug, slot, r, "read", "r"))
+	}
+	for _, w := range iface.path(slot, "write") {
+		fmt.Fprintln(snippet, appArmorEntry(plug, slot, w, "write", "w"))
+	}
+	for _, e := range iface.path(slot, "execute") {
+		fmt.Fprintln(snippet, appArmorEntry(plug, slot, e, "exec", "ixr"))
+	}
+	return snippet.Bytes(), nil
+}
+
+func (iface *OvermountInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityMount:
+		return iface.overmountMountSnippet(plug, slot)
+	case interfaces.SecurityAppArmor:
+		return iface.overmountAppArmorSnippet(plug, slot)
+	}
+	return nil, nil
+}
+
+func (iface *OvermountInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *OvermountInterface) AutoConnect(plug *interfaces.Plug, slot *interfaces.Slot) bool {
+	return plug.Attrs["overmount"] == slot.Attrs["overmount"]
```

Comment:
This logic from the _content_ interface doesn't make sense here. We're not matching content from one snap to the other. This interface is rather asking the system to do certain things to itself instead.